gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/EntityInteractionsFetcher.java

```diff
@@ -539,37 +533,47 @@ private void parseResultSet(
 
   private void addInteractionEdges(
       EntityInteraction.Builder interaction,
-      Set<String> selectedColumns,
+      Collection<Expression> selections,
       String fromEntityType,
       EntityKey fromEntityId,
       String toEntityType,
       EntityKey toEntityId) {
 
-    if (selectedColumns.contains(FROM_ENTITY_ID_ATTRIBUTE_ID)) {
+    Map<String, String> selectionResultNames =
+        selections.stream()
+            .filter(ExpressionReader::isSimpleAttributeSelection)
+            .collect(
+                Collectors.toUnmodifiableMap(
+                    expression ->
+                        ExpressionReader.getSelectionAttributeId(expression).orElseThrow(),
+                    expression ->
+                        ExpressionReader.getSelectionResultName(expression).orElseThrow()));
+
+    if (selectionResultNames.containsKey(FROM_ENTITY_ID_ATTRIBUTE_ID)) {
       interaction.putAttribute(
-          FROM_ENTITY_ID_ATTRIBUTE_ID,
+          selectionResultNames.get(FROM_ENTITY_ID_ATTRIBUTE_ID),
```

Comment:
Prior, the alias was not used, was that taken care of some other place, or did this change fix the case of not handling alias?

Reply:
Correct, aliases were not working here before this change (well more accurately, graphql was using attribute IDs as aliases, so they matched).